src/domains/profile/pages/Welcome/Welcome.tsx

```diff
@@ -20,7 +20,9 @@ export const Welcome = () => {
 	const [profiles, setProfiles] = useState([]);
 
 	useEffect(() => {
-		setProfiles(env.profiles().all());
+		env.profiles()
+			.all()
+			.then((profiles: any) => setProfiles(profiles));
```

Comment:
Why would a sync method have to use then? https://github.com/ArkEcosystem/platform-sdk/blob/master/packages/platform-sdk-profiles/src/repositories/profile-repository.ts#L30-L32

Reply:
Yeah are you correct. I was just looking at the story https://github.com/ArkEcosystem/desktop-wallet/blob/3.0-react/src/domains/profile/pages/Welcome/Welcome.stories.tsx#L18-L30. So for this case the promise will always log pending as long as its results are not resolved yet.